src/processor/operator/persistent/index_builder.cpp

```diff
@@ -99,25 +120,44 @@ void IndexBuilderSharedState::quitProducer() {
     }
 }
 
-void IndexBuilder::insert(const ColumnChunkData& chunk, offset_t nodeOffset, offset_t numNodes,
-    NodeBatchInsertErrorHandler& errorHandler) {
+namespace {
+OptionalWarningSourceData getWarningDataFromChunks(
+    const std::optional<std::vector<storage::ColumnChunkData*>>& chunks, common::idx_t posInChunk) {
+    OptionalWarningSourceData ret;
+    if (chunks.has_value()) {
+        KU_ASSERT(chunks->size() == CopyConstants::WARNING_METADATA_NUM_COLUMNS);
+        ret = WarningSourceData{chunks.value()[0]->getValue<uint64_t>(posInChunk),
+            chunks.value()[1]->getValue<uint64_t>(posInChunk),
+            chunks.value()[2]->getValue<uint32_t>(posInChunk),
+            chunks.value()[3]->getValue<uint64_t>(posInChunk),
+            chunks.value()[4]->getValue<uint32_t>(posInChunk)};
```

Comment:
should probably add comment saying that these types should match those in `WarningSourceData`

Reply:
could probably also use `decltype` here